pkgs/development/python-modules/sfepy/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, fetchurl
+, numpy
+, scipy
+, matplotlib
+, pyparsing
+, tables
+, cython
+, python
+, sympy
+}:
+
+buildPythonPackage rec {
+  name = "sfepy_${version}";
+  version = "2019.2";
+
+  src = fetchurl {
+    url="https://github.com/sfepy/sfepy/archive/release_${version}.tar.gz";
+    sha256 = "17dj0wbchcfa6x27yx4d4jix4z4nk6r2640xkqcsw0mf62x5l1pj";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+    cython
+    scipy
+    matplotlib
+    pyparsing
+    tables
+    sympy
+  ];
+
+  checkPhase = ''
+    ${python.interpreter} setup.py build_ext --inplace
+    ${python.interpreter} -c "import sfepy"
+    ${python.interpreter} tests/test_assembling.py
```

Comment:
I'm not sure these are actually achieving anything. All any of them do is define a class but nothing ever executes any of the test methods.

On top of this, they fail for me too because importing `sfepy` tries to write to a dotfile in the home directory, which is defined as an unwritable location in sandboxed environments. To get past that error you need to add e.g. `export HOME=$TMPDIR` at the beginning of the `checkPhase`, but even once you've done this I don't think the tests are actually getting executed.

Reply:
Ok, tests WFM with simply:

```
  checkPhase = ''
    export HOME=$TMPDIR
    mv sfepy sfepy.hidden
    ${python.interpreter} run_tests.py
  '';
```

To get the tests to pass on my linux machine I need to also add

```  
  postPatch = ''
    rm tests/test_homogenization_perfusion.py
  '';
```

otherwise this test hangs indefinitely. However I don't know whether this is simply a symptom of my linux machine being slower than the mac, so it's probably worth disabling that test on all platforms.

Reply:
Sorry, my solution was daft. Thanks for fixing it. I did go through the same process and eliminated `test_homogenization_perfusion.py`, however, It seemed like the entire test suite required minutes rather than seconds so I tried to just use a small random subset of the tests (which I might revisit). Should the tests for each package in Nixpkgs be more of a sanity check rather than an entire long running test suite?

Reply:
There _is_ an argument for that, but I wouldn't worry too much, we quite frequently run full test suites and people are ok with it. Try building the `pytest` package for instance - the tests take a while. And when it comes to build times in general, python packages barely come close to how long some c++ packages take, so we've got some headroom to spare.

It's also a bit less maintainable to be listing lots of specific files in the `checkPhase`. There's a degree we want to be able to attempt to bump this package with minimal intervention and use the tests to be happy that the result is valid. If upstream refactors its tests significantly or just renames some of the files mentioned, it would unnecessarily break.

Reply:
Agree, all good points. Thanks for the feedback.